### PR TITLE
Fix mets server: clean leftover zombie processes

### PR DESCRIPTION
**Pull request by MehmedGIT**
This PR fixes:
- No more zombie processes for Mets Servers when they terminate!
- Appropriately deleting the Unix Domain Socket in more complicated configurations (e.g., UDS over TCP multiplexing)
- More extensive logging (to separate files as well) for:
  + `ocrd_network.server_cache.locked_pages`
  + `ocrd_network.server_cache.processing_requests`
  + `ocrd.models.ocrd_mets.server.{self.url}`
  + `ocrd_network.tcp_to_uds_mets_proxy`
  
Please make sure you adapt your logging configuration file appropriately to avoid extreme amounts of logs. Maybe an adaption is needed to the default logging configuration file - open for discussions.

### CODE REVIEW

**Comment by MehmedGIT** [comment]
@joschrew, would be great if you could test your docker setup with this PR to see if there are any issues before this is merged.

File: src/ocrd/mets_server.py
Comment (kba) [inline src/ocrd/mets_server.py:442]:
```suggestion
```

Graceful shutdown working nicely AFAICT, so the this can go for good.

File: src/ocrd/mets_server.py
Comment (kba) [inline src/ocrd/mets_server.py:452]:
```suggestion
        self.log.info(f"Configuring the Mets Server")
```

Though I found `Starting up` clearer.

File: src/ocrd/mets_server.py
Comment (kba) [inline src/ocrd/mets_server.py:598]:
```suggestion
```

Then let's remove it, it was a workaround anyway.

File: src/ocrd/mets_server.py
Comment (kba) [inline src/ocrd/mets_server.py:613]: Cannot propose outside the diff context, but if we don't need it any more, feel free to rremove `def is_socket_in_use...`.

File: src/ocrd_network/runtime_data/deployer.py
Comment (kba) [inline src/ocrd_network/runtime_data/deployer.py:194]:
```suggestion
            self.log.info(f"Mets server with pid: {mets_server_pid} has already terminated.")
```

**Review by kba — CHANGES_REQUESTED** [review]
Excellent, this will make managing the PS much easier.

`_exit` was a terrible idea, obviously, my bad.

Besides some minor documentation issues, LGTM.

However, I am unsure about the additional logging. Not so much the new logs (which are appreciated) but the upgrading from `log.debug` to `log.info`. Since at least in my use case (kubernetes), everything, including all separate log files, is redirected to a single STDOUT stream, this will make it hard to spot relevant logging in all the noise. At least we should use different loggers for the very frequent loggings (i.e. have a `self.log/ocrd_network.processing_server` for regular stuff and `self.verbose_log/ocrd_network.processing_server_verbose`), so it is easier to configure in the `ocrd_logging.conf` and to filter by in the log viewer?

File: src/ocrd/mets_server.py
Comment (MehmedGIT) [reply src/ocrd/mets_server.py:452]: Yes, but then there is another log right before starting the uvicorn server at the end of the method. I thought a differentiation between the two may be good.

File: src/ocrd/mets_server.py
Comment (MehmedGIT) [reply src/ocrd/mets_server.py:598]: Yes, but @joschrew should potentially first confirm that with his docker setup. I did not want to remove that since he added the workaround and I could not test with his setup.

**Review by kba — APPROVED** [review]
> However, I am unsure about the additional logging. Not so much the new logs (which are appreciated) but the upgrading from `log.debug` to `log.info`. Since at least in my use case (kubernetes), everything, including all separate log files, is redirected to a single STDOUT stream, this will make it hard to spot relevant logging in all the noise. At least we should use different loggers for the very frequent loggings (i.e. have a `self.log/ocrd_network.processing_server` for regular stuff and `self.verbose_log/ocrd_network.processing_server_verbose`), so it is easier to configure in the `ocrd_logging.conf` and to filter by in the log viewer?

The logging is already distributed over different loggers, so that part is not an issue.

And the `debug->info` I can also live with, the intention is to not force users to set global log level to `debug` to see these. In any case, it can be adapted in the logging.conf if desired.

I'll be deploying this now and it is good to merge once @joschrew okays it.

**Comment by MehmedGIT** [comment]
> There the socket-file is deleted in Deployer.start_uds_mets_server when starting the mets_server for a workspace and the socket-file is existing. This is necessary for me to work because when shutting down the docker containers the socket file is not deleted. It seems to me, that the shutdown Method of OcrdMetsServer is not called, I couldn't find any log entry, which should have been issued in this method. But that's ok for everything to work for me.

Strangely, it seems the DELETE request is never sent to the mets server in the docker environment. The socket file is volume mapped on the host OS, and deleted with a shutdown method call by the Mets server process that resides inside a container. So the socket file would not be removed even if the shutdown method was invoked correctly. It seems that not much can be done for the docker environment other than the fallback [solution](https://github.com/OCR-D/core/blob/network_client_block_prints/src/ocrd_network/runtime_data/deployer.py#L149-L153) you talked about.

File: src/ocrd/mets_server.py
Comment (MehmedGIT) [reply src/ocrd/mets_server.py]: Resolved with [5055309](https://github.com/OCR-D/core/pull/1284/commits/50553093180ac6b641273c08d559bbadf5e9b1d2)

**Pull request merged**
